src/components/SplitVertical/SplitVertical.jsx

```diff
@@ -0,0 +1,392 @@
+			secondary,
+			right,
+			primaryRef,
+		} = this.panes;
+
+		this.applyDeltaToSecondaryWidth(dX, isExpanded, this.secondaryStartRect, secondaryRef, secondary, right, this.refs.inner, primaryRef, collapseShift);
+	},
+
+	handleDragEnd({ dX }, { event }) {
+		const {
+			isExpanded,
+			collapseShift,
+			onResize,
+		} = this.props;
+
+		const {
+			secondaryRef,
+			secondary,
+			right,
+			primaryRef,
+		} = this.panes;
+
+		onResize(
+			this.applyDeltaToSecondaryWidth(dX, isExpanded, this.secondaryStartRect, secondaryRef, secondary, right, this.refs.inner, primaryRef, collapseShift),
+			{ props: this.props, event }
+		);
+
+		this.resetAnimation(this.refs.inner, secondaryRef, primaryRef);
+	},
+
+	componentWillReceiveProps(nextProps) {
+		const {
+			isAnimated,
+			isExpanded,
+			collapseShift,
+		} = nextProps;
+
+		const {
+			primaryRef,
+			secondaryRef,
+			secondary,
+			right,
+		} = this.getPanes();
+
+		if (this.props.isExpanded && !isExpanded) { // collapse secondary
+			const secondaryRect = secondaryRef.getBoundingClientRect();
+			this.collapseSecondary(this.refs.inner, secondary, right, primaryRef, secondaryRect.width - collapseShift);
+		} else if (!this.props.isExpanded && isExpanded) { // expand secondary
+			this.expandSecondary(this.refs.inner, secondary, right, primaryRef);
+		}
+
+		if (this.state.isAnimated !== isAnimated) {
+			this.setState({
+				isAnimated,
+			});
+		}
+	},
+
+	componentDidMount() {
+		const {
+			isExpanded,
+			isAnimated,
+			collapseShift,
+		} = this.props;
+
+		const {
+			primaryRef,
+			secondaryRef,
+			secondary,
+			right,
+		} = this.getPanes();
+
+		const {
+			inner,
+		} = this.refs;
+
+		_.defer(() => {
+			if (!isExpanded) { // collapse secondary
+				const secondaryRect = secondaryRef.getBoundingClientRect();
+				this.collapseSecondary(inner, secondary, right, primaryRef, secondaryRect.width - collapseShift);
+			}
+			this.disableAnimation(inner, secondaryRef, primaryRef);
+
+			_.defer(() => {
+				if (isAnimated) {
+					this.setState({ isAnimated });
+				}
+				this.resetAnimation(inner, secondaryRef, primaryRef);
+			});
+		});
+	},
+
+	render() {
+		const {
+			children,
+			className,
+			isExpanded,
+			...passThroughs,
+		} = this.props;
+
+		const {
+			isAnimated,
+		} = this.state;
+
+		const {
+			left: leftPaneProps,
+			right: rightPaneProps,
+			secondary,
+		} = this.getPanes();
+
+		const dividerProps = _.get(_.first(filterTypes(children, SplitVertical.Divider)), 'props', {});
+
+		return (
+			<div
+				{...omitProps(passThroughs, SplitVertical)}
+				className={cx('&', {
+					'&-is-expanded': isExpanded,
+					'&-is-animated': isAnimated,
+				}, className)}
+			>
+				<div className={cx('&-inner')} ref='inner'>
+					<div
+						{...omitProps(leftPaneProps, SplitVertical.LeftPane)}
+						className={cx('&-LeftPane', {
+							'&-is-secondary': leftPaneProps === secondary,
+						}, leftPaneProps.className)}
+						style={_.assign({}, {
```

Comment:
not sure why i'm using `_.assign` here rather than the spread operator, but i'll try to switch it up